modules/utxo-lib/src/bitgo/psbt/fromHalfSigned.ts

```diff
@@ -1,145 +1,86 @@
-import { PartialSig, PsbtInputUpdate } from 'bip174/src/lib/interfaces';
-import * as opcodes from 'bitcoin-ops';
-import { Stack, ScriptSignature, TxOutput, ecc as eccLib, script as bscript, taproot, payments } from '../..';
+import { PsbtInputUpdate, PartialSig } from 'bip174/src/lib/interfaces';
+import { ecc as eccLib, TxOutput, taproot } from '../..';
 import { UtxoTransaction } from '../UtxoTransaction';
-import { isP2wpkh, isP2wsh, isTaproot } from './scriptTypes';
+import { parseSignatureScript } from '../parseInput';
+import { getSignaturesWithPublicKeys } from '../signature';
+import { hasWitnessData } from '../outputScripts';
 
-/**
- * @param hashFn
- * @param script
- * @param stack
- *
- * @return partial signatures for a 2of3 p2ms script
- */
-function matchSignatures(hashFn: (hashType: number) => Buffer, script: Buffer, stack: Stack): PartialSig[] {
-  const partialSig: PartialSig[] = [];
-  const publicKeys = payments.p2ms({ output: script }).pubkeys;
-  if (!publicKeys || publicKeys.length !== 3) {
-    throw new Error('Invalid multisig script');
-  }
-  for (const sig of stack.slice(1)) {
-    // Ignore extra empty element re MULTISIG bug
-    if (!Buffer.isBuffer(sig) || sig.length === 0) continue;
-    const { signature, hashType } = ScriptSignature.decode(sig);
-    const hash = hashFn(hashType);
-    let signatureMatched = false;
-    for (const pubkey of publicKeys) {
-      if (eccLib.verify(hash, pubkey, signature)) {
-        signatureMatched = true;
-        partialSig.push({ pubkey, signature: sig });
-        break;
-      }
-    }
-    if (!signatureMatched) {
-      throw new Error('Invalid signature in partially signed transaction');
-    }
-  }
-  return partialSig;
+function omitUndefined<T extends Record<string, unknown>>(v: Record<string, unknown>): T {
+  return Object.fromEntries(Object.entries(v).filter(([k, v]) => v !== undefined)) as T;
 }
 
 export function getInputUpdate(
   tx: UtxoTransaction<bigint>,
   vin: number,
-  prevOut: TxOutput<bigint> & { prevTx?: Buffer }
+  prevOuts: (TxOutput<bigint> & { prevTx?: Buffer })[]
 ): PsbtInputUpdate {
-  const input = tx.ins[vin];
-  const update: PsbtInputUpdate = {};
-  if (prevOut.prevTx) {
-    update.nonWitnessUtxo = prevOut.prevTx;
+  const nonWitnessUtxo = prevOuts[vin].prevTx;
```

Comment:
How do we know this is a non-witness utxo? I realize this is old logic being refactored but I couldn't make sense of it.